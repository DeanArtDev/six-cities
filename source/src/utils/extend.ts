const extend = (a: object, b: object): object => {
  const emptyObj: object = {};

  return Object.assign(emptyObj, a, b);
};

export default extend;
