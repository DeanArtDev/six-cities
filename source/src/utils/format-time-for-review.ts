import {MONTHS} from '@root/consts/consts';

const formatTimeForReview = (date: string): string => {
  const nextDate = new Date(date);
  const month = MONTHS[nextDate.getMonth()];
  const day = nextDate.getDate() < 10 ? `0${nextDate.getDate()}` : nextDate.getDate();

  return `${month} ${day} ${nextDate.getFullYear()}`;
};

export default formatTimeForReview;
