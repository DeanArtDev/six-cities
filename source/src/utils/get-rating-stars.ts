const ONE_STAR_PRESENTS = 20;

const getRatingStars = (value: number): number | 0 => {
  if (value === 0 || value === undefined || isNaN(value)) {
    return 0;
  }

  return Math.ceil(value) * ONE_STAR_PRESENTS;
};

export default getRatingStars;
