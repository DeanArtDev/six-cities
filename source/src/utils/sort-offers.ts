import {OffersSortTypes} from '@root/types';

const sortOffers = (typeSort, offers) => {
  switch (typeSort) {
    case OffersSortTypes.LOW_TO_HIGH:
      return offers
        .slice()
        .sort((max, min) => max.price - min.price);

    case OffersSortTypes.HIGH_TO_LOW:
      return offers
        .slice()
        .sort((max, min) => min.price - max.price);

    case OffersSortTypes.RATED_FIRST:
      return offers
        .slice()
        .sort((max, min) => min.rating - max.rating);

    default:
      return offers;
  }
};

export default sortOffers;
