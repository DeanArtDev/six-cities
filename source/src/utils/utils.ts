export {default as getRatingStars} from './get-rating-stars';
export {default as extend} from './extend';
export {default as sortOffers} from './sort-offers';
export {default as formatTimeForReview} from './format-time-for-review';
export {default as noop} from './noop';
