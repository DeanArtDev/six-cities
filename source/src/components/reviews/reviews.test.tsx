import {WrapperComponent as Reviews} from './reviews';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {TestMock} from '@root/test-mock/test-mock';
import {Router} from 'react-router-dom';
import history from '@root/history';
import {noop} from '@root/utils/utils';


describe(`<Reviews/>: `, () => {

  describe(`Authorization: `, () => {
    it(`should render AUTH currently`, () => {
      const tree = renderer
        .create(
            <Router
              history={history}
            >
              <Reviews
                comments={TestMock.adaptedComments}
                onFormSubmit={noop}
                isSend={false}
                isLoading={false}
                isAuth={false}
                errors={undefined}
                reviewValue={`e`}
                onRatingChange={noop}
                onReviewChange={noop}
                ratingRef={React.createRef()}
              />
            </Router>
        ).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it(`should render NOT AUTH currently`, () => {
      const tree = renderer
        .create(
            <Router
              history={history}
            >
              <Reviews
                comments={TestMock.adaptedComments}
                onFormSubmit={noop}
                isSend={false}
                isLoading={false}
                isAuth={false}
                errors={undefined}
                reviewValue={`e`}
                onRatingChange={noop}
                onReviewChange={noop}
                ratingRef={React.createRef()}
              />
            </Router>
        ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`App Store: `, () => {
    it(`should render READY state currently`, () => {
      const tree = renderer
      .create(
          <Router
            history={history}
          >
            <Reviews
              comments={TestMock.adaptedComments}
              onFormSubmit={noop}
              isSend={false}
              isLoading={false}
              isAuth={true}
              errors={undefined}
              reviewValue={`e`}
              onRatingChange={noop}
              onReviewChange={noop}
              ratingRef={React.createRef()}
            />
          </Router>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it(`should render LOADING state currently`, () => {
      const tree = renderer
      .create(
          <Router
            history={history}
          >
            <Reviews
              comments={TestMock.adaptedComments}
              onFormSubmit={noop}
              isSend={false}
              isLoading={true}
              isAuth={true}
              errors={undefined}
              reviewValue={`e`}
              onRatingChange={noop}
              onReviewChange={noop}
              ratingRef={React.createRef()}
            />
          </Router>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it(`should render SEND state currently`, () => {
      const tree = renderer
      .create(
          <Router
            history={history}
          >
            <Reviews
              comments={TestMock.adaptedComments}
              onFormSubmit={noop}
              isSend={true}
              isLoading={false}
              isAuth={true}
              errors={undefined}
              reviewValue={`e`}
              onRatingChange={noop}
              onReviewChange={noop}
              ratingRef={React.createRef()}
            />
          </Router>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`Errors: `, () => {
    it(`should render Error currently`, () => {
      const tree = renderer
        .create(
            <Router
              history={history}
            >
              <Reviews
                comments={TestMock.adaptedComments}
                onFormSubmit={noop}
                isSend={false}
                isLoading={false}
                isAuth={true}
                errors={`error`}
                reviewValue={`e`}
                onRatingChange={noop}
                onReviewChange={noop}
                ratingRef={React.createRef()}
              />
            </Router>
        ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});

