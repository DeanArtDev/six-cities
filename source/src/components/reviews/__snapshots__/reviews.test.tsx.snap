// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews/>:  App Store:  should render LOADING state currently 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mollie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Dean
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        className="form__rating-input visually-hidden"
        disabled={true}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={true}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={true}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={true}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={true}
        id="1-star"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-star"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={true}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value="e"
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set
        <span
          className="reviews__star"
        >
          rating
        </span>
        and describe your stay with at least
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        type="submit"
      >
        Loading...
      </button>
    </div>
  </form>
</section>
`;

exports[`<Reviews/>:  App Store:  should render READY state currently 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mollie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Dean
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="1-star"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-star"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={false}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value="e"
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set
        <span
          className="reviews__star"
        >
          rating
        </span>
        and describe your stay with at least
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;

exports[`<Reviews/>:  App Store:  should render SEND state currently 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mollie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Dean
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="1-star"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-star"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={false}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value="e"
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set
        <span
          className="reviews__star"
        >
          rating
        </span>
        and describe your stay with at least
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;

exports[`<Reviews/>:  Authorization:  should render AUTH currently 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mollie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Dean
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
  </ul>
</section>
`;

exports[`<Reviews/>:  Authorization:  should render NOT AUTH currently 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mollie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Dean
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
  </ul>
</section>
`;

exports[`<Reviews/>:  Errors:  should render Error currently 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mollie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Dean
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location
        </p>
        <time
          className="reviews__time"
          dateTime="2020-02-13"
        >
          February 13 2020
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label error-message"
      htmlFor="review"
    >
      error
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="1-star"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-star"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={false}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value="e"
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set
        <span
          className="reviews__star"
        >
          rating
        </span>
        and describe your stay with at least
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
